### PR TITLE
COMMONS-487: Add owasp-java-html-sanitizer library to encode data in …

### PR DESCRIPTION
**Pull request by kmenzli**
…presentation layer

### CODE REVIEW

File: commons-component-common/src/main/java/org/exoplatform/commons/utils/HTMLSanitizer.java [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+ * OWASP Java HTML Sanitizer API is a library (not a filter software) that provides XSS/XEE protection
+ * as an API to encode data in presentation layer.
+ * Whenever something that is affected from user input is to be displayed, proper encoding should be applied.
+ * <p>
+ * Moreover, we aim to replace the old lib called AntiSamy by this one
+ * applications relying on this through to the end of lifecycle.
+ *
+ * @author <a href="kmenzli@exoplatform.com">Khemais MENZLI</a>
+ * @version $Revision$
+ */
```
Comment (thomasdelhomenie) [inline commons-component-common/src/main/java/org/exoplatform/commons/utils/HTMLSanitizer.java]: you should describe what the class does instead.

File: commons-component-common/src/test/java/org/exoplatform/commons/utils/HTMLSanitizerTest.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+package org.exoplatform.commons.utils;
+
+import org.exoplatform.commons.testing.BaseCommonsTestCase;
+import org.junit.Test;
+import org.owasp.html.examples.EbayPolicyExample;
+
+/**
+ * Created by kmenzli on 16/06/16.
+ */
+public class HTMLSanitizerTest extends BaseCommonsTestCase {
```
Comment (thomasdelhomenie) [inline commons-component-common/src/test/java/org/exoplatform/commons/utils/HTMLSanitizerTest.java]: Why not using the JUnit 4 way, with annotations ? Do you really need to extend from BaseCommonsTestCase ?

File: commons-component-common/src/test/java/org/exoplatform/services/deployment/UtilsTest.java [hunk 3]
```diff
@@ -165,17 +165,11 @@ public void testImportNode() throws Exception {
...
-    assertEquals(Utils.sanitize("12345 78 0"), "12345 78 0");
-    assertEquals(Utils.sanitize("12345 78 012345 78 012345 78 012345 78 012345 78 012345 78 012345 78 012345 78 012345 78 0"), 
-    	  "12345 78 012345 78 012345 78 012345 78 012345 78 012345 78 012345 78 012345 78 012345 78 0");
-  }
-  
   /**
    * Export a node to XML with document view or system view
    * 
    * @param currentNode
-   * @param docview or sysview
```
Comment (thomasdelhomenie) [inline commons-component-common/src/test/java/org/exoplatform/services/deployment/UtilsTest.java]: Why this change ?

Comment (kmenzli) [inline commons-component-common/src/test/java/org/exoplatform/services/deployment/UtilsTest.java]: The old param name is not correct and it doesn't match with method param

Comment (thomasdelhomenie) [inline commons-component-common/src/test/java/org/exoplatform/services/deployment/UtilsTest.java]: so it should be "@param format docview or sysview"

Comment (thomasdelhomenie) [inline commons-component-common/src/test/java/org/exoplatform/services/deployment/UtilsTest.java]: Yes of course, it is even better :)

File: commons-component-common/src/test/java/org/exoplatform/commons/utils/HTMLSanitizerTest.java [hunk 4]
```diff
@@ -18,25 +18,27 @@
...
 import org.junit.Test;
-import org.owasp.html.examples.EbayPolicyExample;
 
 /**
  * Created by kmenzli on 16/06/16.
  */
-public class HTMLSanitizerTest extends BaseCommonsTestCase {
+public class HTMLSanitizerTest {
 
+    @Test(timeout=100)
```
Comment (thomasdelhomenie) [inline commons-component-common/src/test/java/org/exoplatform/commons/utils/HTMLSanitizerTest.java]: I don't think adding a timeout is necessary on all tests. That could be interesting if the test uses external resources, with connections or whatever, but in this class it is not the case, so they can be removed IMO.

Comment (kmenzli) [inline commons-component-common/src/test/java/org/exoplatform/commons/utils/HTMLSanitizerTest.java]: I thought that I should use 10millisecond instead on 100 :)
IMO, RT between adding a comment/tweet and the sanitization shall not exceed 100ms, thus I prefer to keep this timeout test. 
From a threshold standpoint I think we have to keep this timeout param?

Comment (thomasdelhomenie) [inline commons-component-common/src/test/java/org/exoplatform/commons/utils/HTMLSanitizerTest.java]: Why 100 or 10, and not 50 ? or 5 ?
Timeout is not for performance tests IMO. The doc mentions : "Warning: while timeout is useful to catch and terminate infinite loops, it should not be considered deterministic. The following test may or may not fail depending on how the operating system schedules threads"

Comment (nghinv) [inline commons-component-common/src/test/java/org/exoplatform/commons/utils/HTMLSanitizerTest.java]: must remove antisamy here: https://github.com/exodev/commons/blob/fix/COMMONS-487/commons-component-common/pom.xml#L92

Comment (kmenzli) [inline commons-component-common/src/test/java/org/exoplatform/commons/utils/HTMLSanitizerTest.java]: @nghinv good catch
